staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4298,6 +4298,13 @@ type PodStatus struct {
 	// +featureGate=DynamicResourceAllocation
 	// +optional
 	ResourceClaimStatuses []PodResourceClaimStatus `json:"resourceClaimStatuses,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name" protobuf:"bytes,15,rep,name=resourceClaimStatuses"`
+
+	// RequestedResources is the effective resource requests for this pod taking into consideration init
+	// and regular containers, and the pod overhead. This field reflects the total desired resources for
+	// the pod which may differs from actual resources allocated for the pod by the node.
```

Comment:
`may differ`.  

Then one more question - when did we spec that the pod updates this.  After resize is accepted?  Or independent of that, once the kubelet observes the value?  Why?

Depending on that answer, we should add one more sentence: `The node will update this value as it <observes|accepts> changes to requested resources.  If no node can be assigned by the scheduler this value will be set to ease debugging of out-of-resource situations.`

Also, this needs to be feature gated (it's a new field on a v1 object), so should have a comment down below about which feature gate controls it and should only be set when the feature gate is true.

Reply:
> Then one more question - when did we spec that the pod updates this. After resize is accepted? Or independent of that, once the kubelet observes the value? Why?

Its when kubelet observes the value, its called from `generateAPIPodStatus->convertStatusToAPIStatus`.  It looks like resizing the pod does an update of the `.Resize` field in `handlePodResourcesResize`, so I think in the case of sidecar+vertical pod resizing enabled and a pod is being resized, there's a chance it will be out of date for a short period.

Reply:
@tzneal I have not looked deeply into this besides the parts related to in-place resize. IIRC from KEP discussions a few months ago, am I correct in saying that the primary use case for this is for an entity like cluster autoscaler to easily see the pod resource needs in order to provision additional nodes? If yes, can this not be set (protected by featuregate that @smarterclayton pointed out) in apiserver? (PodStrategy.PrepareForCreate much like QOSClass). IMHO that might simplify things with kubelet being the only writer after the initial write.

Reply:
> am I correct in saying that the primary use case for this is for an entity like cluster autoscaler 

I don't think so, the primary use case is users that want to know why pods won't schedule.  Its a bit difficult to look at a pod spec with a mixture of init containers, sidecar init contianers and main containers and know the actual amount of resources required.

That's a good thought in the PrepareForCreate, I think its cleaner and will look into it.